core/src/main/java/org/mskcc/cbio/portal/scripts/ImportResourceDefinition.java

```diff
@@ -0,0 +1,330 @@
+     * 
+     * @param line
+     * @param headerNames
+     * @return the list of values, one for each column. Value will be "" for empty
+     *         columns.
+     */
+    private String[] getFieldValues(String line, String[] headerNames) {
+        // split on delimiter:
+        String[] fieldValues = line.split(DELIMITER, -1);
+
+        // validate: if number of fields is incorrect, give exception
+        if (fieldValues.length != headerNames.length) {
+            throw new IllegalArgumentException("Number of columns in line is not as expected. Expected: "
+                    + headerNames.length + " columns, found: " + fieldValues.length + ", for line: " + line);
+        }
+
+        // now iterate over lines and trim each value:
+        for (int i = 0; i < fieldValues.length; i++) {
+            fieldValues[i] = fieldValues[i].trim();
+        }
+        return fieldValues;
+    }
+
+    private Boolean isValueNotMissing(String value) {
+        if (MissingValues.has(value)) {
+            return false;
+        }
+        return true;
+    }
+
+    private Boolean isValidResourceType(String value) {
+        try {
+            ResourceTypes.valueOf(value);
+        } catch (IllegalArgumentException ex) {
+            throw new IllegalArgumentException("Resource_Type should be one of the following : 'SAMPLE', 'PATIENT' or 'STUDY'."
+                    + "found: " + value);
+        }
+        return true;
+    }
+
+    private Boolean isValidOpenByDefault(String value) {
+        if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false")) {
+            return true;
+        }
+        // open by default value not valid, will set as false by default
+        ProgressMonitor.logWarning("OpenByDefault value is not true or false, set to false by default.");
+        return false;
+    }
+
+    private int findAndValidateResourceIdColumn(String[] headerNames) {
+        return findAndValidateColumnIndexInHeaders(RESOURCE_ID_COLUMN_NAME, headerNames);
+    }
+
+    private int findAndValidateDisplayNameColumn(String[] headerNames) {
+        return findAndValidateColumnIndexInHeaders(DISPLAY_NAME_COLUMN_NAME, headerNames);
+    }
+
+    private int findAndValidateDescriptionColumn(String[] headerNames) {
+        return findAndValidateColumnIndexInHeaders(DESCRIPTION_COLUMN_NAME, headerNames);
+    }
+
+    private int findAndValidateResourceTypeColumn(String[] headerNames) {
+        return findAndValidateColumnIndexInHeaders(RESOURCE_TYPE_COLUMN_NAME, headerNames);
+    }
+
+    private int findAndValidateOpenByDefaultColumn(String[] headerNames) {
+        return findAndValidateColumnIndexInHeaders(OPEN_BY_DEFAULT_COLUMN_NAME, headerNames);
+    }
+
+    private int findAndValidatePriorityColumn(String[] headerNames) {
+        return findAndValidateColumnIndexInHeaders(PRIORITY_COLUMN_NAME, headerNames);
+    }
+
+    private int findAndValidateColumnIndexInHeaders(String columnHeader, String[] headerNames) {
```

Comment:
could we send `Map<String,Integer>` of `headerNames` instead of array? checking in map is more efficient than iterating an array